formats/cbor/commonMain/src/kotlinx/serialization/cbor/internal/CborElementSerializers.kt

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright 2017-2025 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+@file:OptIn(ExperimentalSerializationApi::class, ExperimentalUnsignedTypes::class)
+
+package kotlinx.serialization.cbor.internal
+
+import kotlinx.serialization.*
+import kotlinx.serialization.builtins.*
+import kotlinx.serialization.cbor.*
+import kotlinx.serialization.descriptors.*
+import kotlinx.serialization.encoding.*
+
+/**
+ * Serializer object providing [SerializationStrategy] and [DeserializationStrategy] for [CborElement].
+ * It can only be used by with [Cbor] format and its input ([CborDecoder] and [CborEncoder]).
+ */
+internal object CborElementSerializer : KSerializer<CborElement> {
+    override val descriptor: SerialDescriptor =
+        buildSerialDescriptor("kotlinx.serialization.cbor.CborElement", PolymorphicKind.SEALED) {
+            // Resolve cyclic dependency in descriptors by late binding
+            element("CborPrimitive", defer { CborPrimitiveSerializer.descriptor })
+            element("CborNull", defer { CborNullSerializer.descriptor })
+            element("CborString", defer { CborStringSerializer.descriptor })
+            element("CborBoolean", defer { CborBooleanSerializer.descriptor })
+            element("CborByteString", defer { CborByteStringSerializer.descriptor })
+            element("CborMap", defer { CborMapSerializer.descriptor })
+            element("CborList", defer { CborListSerializer.descriptor })
+            element("CborDouble", defer { CborDoubleSerializer.descriptor })
+            element("CborInt", defer { CborIntSerializer.descriptor })
+            element("CborUInt", defer { CborUIntSerializer.descriptor })
+        }
+
+    override fun serialize(encoder: Encoder, value: CborElement) {
+        encoder.asCborEncoder()
```

Comment:
Ideally you allow serializing to other formats with the "normal" behaviour (see the serializers for `JsonElement` etc.)

Reply:
doesn't the JsonElement serializer have a `verify` method that ensures it can only be used with the JSON format?

Reply:
turns out: [it does](https://github.com/Kotlin/kotlinx.serialization/blob/10c7b9d2d6a333aee66c15c2f8d224b8e7c71660/formats/json/commonMain/src/kotlinx/serialization/json/JsonElementSerializers.kt#L193)

Reply:
@JesusMcCloud You are right. I had missed the check because it was encapsulated. The worst part is that it would actually work without that verify check.

Reply:
it does make life easier for those implementing the serializers. for me at least, i don't have to think about how this will behave with other formats, esp. wrt. tagging.